File: AspNetMvcUnleashed/0672329980_source/Chapter20/VB/UnleashedBlog/UnleashedBlog.Tests/Factories/BlogEntryFactory.vb

Public Class BlogEntryFactory

    Public Shared Function [Get]() As BlogEntry
        Dim blogEntry = New BlogEntry()
        blogEntry.Title = "Test Entry"
        blogEntry.Name = "Test Entry"
        blogEntry.Text = "Blog text"
        blogEntry.DatePublished = New DateTime(2010, 12, 25)
        Return blogEntry
    End Function

    Public Shared Function GetWithDatePublished(ByVal datePublished As DateTime) As BlogEntry
        Dim blogEntry = [Get]()
        blogEntry.DatePublished = datePublished
        Return blogEntry
    End Function

    Public Shared Function GetWithTitle(ByVal title As String) As BlogEntry
        Dim blogEntry = [Get]()
        blogEntry.Title = title
        Return blogEntry
    End Function


    Public Shared Function GetWithName(ByVal name As String) As BlogEntry
        Dim blogEntry = [Get]()
        blogEntry.Name = name
        Return blogEntry
    End Function


    Public Shared Function GetWithText(ByVal text As String) As BlogEntry
        Dim blogEntry = [Get]()
        blogEntry.Text = text
        Return blogEntry
    End Function


End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter19/CS/UnleashedBlog/UnleashedBlog.Tests/Factories/BlogEntryFactory.cs
using System;
using UnleashedBlog.Models;

namespace UnleashedBlog.Tests
{
    public class BlogEntryFactory
    {

        public static BlogEntry Get()
        {
            var blogEntry = new BlogEntry();
            blogEntry.Title = "Test Entry";
            blogEntry.Name = "Test Entry";
            blogEntry.Text = "Blog text";
            blogEntry.DatePublished = new DateTime(2010, 12, 25);
            return blogEntry;
        }

        public static BlogEntry GetWithDatePublished(DateTime datePublished)
        {
            var blogEntry = Get();
            blogEntry.DatePublished = datePublished;
            return blogEntry;
        }

        public static BlogEntry GetWithTitle(string title)
        {
            var blogEntry = Get();
            blogEntry.Title = title;
            return blogEntry;
        }


        public static BlogEntry GetWithName(string name)
        {
            var blogEntry = Get();
            blogEntry.Name = name;
            return blogEntry;
        }


        public static BlogEntry GetWithText(string text)
        {
            var blogEntry = Get();
            blogEntry.Text = text;
            return blogEntry;
        }


    }
}
